app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "name": error["code"],
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, errors) for key,
+        errors in linter_report.items()
```

Comment:
```suggestion
        format_single_linter_file(key, errors) 
        for key, errors in linter_report.items()
```
key - is bad variable name. In this case rename to file_path